libs/SalesforceSDK/src/com/salesforce/androidsdk/security/SalesforceKeyGenerator.java

```diff
@@ -171,10 +199,10 @@ private synchronized static String generateEncryptionKey(String name) {
         try {
             final String keyString = getUniqueId(name);
             byte[] secretKey = keyString.getBytes(StandardCharsets.UTF_8);
-            final MessageDigest md = MessageDigest.getInstance(SHA1);
+            final MessageDigest md = MessageDigest.getInstance(SHA256);
             secretKey = md.digest(secretKey);
-            byte[] dest = new byte[16];
-            System.arraycopy(secretKey, 0, dest, 0, 16);
+            byte[] dest = new byte[32];
```

Comment:
Why are we changing upgradeTo7Dot1, shouldn't the move from 128 to 256 happen in upgradeTo8Dot2?